### ISSUE TITLE
Fit_Generator and predict yielding different outputs

### ISSUE THREAD

**Issue by aditsanghvi94** [keras-team/keras#7877]
I have 3 classes. Class 0 : 50% of data, 1:30% ,2:20%. Total examples are 800,400 train 200 val, 200 test. Each image belongs to only 1 class.

I am training a very simple convnet for multiple class classification
```
model = Sequential()
model.add(Conv2D(16, (3, 3), input_shape=input_shape))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Conv2D(32, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Conv2D(64, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))


model.add(Conv2D(128, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Flatten())
model.add(Dense(64))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(32))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(3))
model.add(Activation('sigmoid'))

model.compile(loss='categorical_crossentropy',
              optimizer='rmsprop',
              metrics=['accuracy'])

```

I am using imagegenerator to augment images. Created dir structure correctly.

```
datagen = ImageDataGenerator(rotation_range= 0.360,
                                 horizontal_flip = True,
                                 vertical_flip = True,
                                 rescale=1. / 255) 

   
img_width, img_height = 350,350
batch_size = 16
generator = datagen.flow_from_directory(
    train_data_dir,
    target_size=(img_width, img_height),
    batch_size=batch_size)
val_generator = datagen.flow_from_directory(
    validation_data_dir,
    target_size=(img_width, img_height),
    batch_size=batch_size)
```

I get an accuracy of 65% training, 65% val at the end.

When i try to predict using my model, I get this output  on any data (test or train as well) for the imgs:

[[  1.00000000e+00   1.00000000e+00   2.31831675e-26]]
[[  1.00000000e+00   1.00000000e+00   4.62001088e-27]]
[[  1.00000000e+00   1.00000000e+00   1.61389520e-27]]
[[  1.00000000e+00   1.00000000e+00   2.85950715e-27]]
[[  1.00000000e+00   1.00000000e+00   4.34395772e-27]]
[[  1.00000000e+00   1.00000000e+00   6.68413629e-25]]
[[  1.00000000e+00   1.00000000e+00   5.57866408e-28]]
[[  1.00000000e+00   1.00000000e+00   2.55804569e-22]]
[[  1.00000000e+00   1.00000000e+00   9.38219268e-26]]
[[  1.00000000e+00   1.00000000e+00   2.81696856e-27]]
[[  1.00000000e+00   1.00000000e+00   3.03054745e-27]]
[[  1.00000000e+00   1.00000000e+00   5.63251077e-27]]
[[  1.00000000e+00   1.00000000e+00   2.44115877e-27]]
[[  1.00000000e+00   1.00000000e+00   3.06371315e-25]]
[[  1.00000000e+00   1.00000000e+00   4.67070906e-27]]
[[  1.00000000e+00   1.00000000e+00   9.45051359e-26]]
[[  1.00000000e+00   1.00000000e+00   1.16351225e-23]]
[[  1.00000000e+00   1.00000000e+00   1.19110686e-26]]

Essentially it predicts 1 for class 0 and class 1.

How is this happening? When i do evaluate generator, it gives me 65% accuracy. Is there a bug in using generators? This predict output is clearly wrong .
